buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.kt

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import com.android.build.gradle.LibraryExtension
+import com.github.sherter.googlejavaformatgradleplugin.GoogleJavaFormatExtension
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.google.firebase.gradle.plugins.ci.device.FirebaseTestServer
+import com.google.firebase.gradle.plugins.license.LicenseResolverPlugin
+import java.io.File
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.tasks.GenerateModuleMetadata
+import org.gradle.kotlin.dsl.getByType
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class FirebaseLibraryPlugin : Plugin<Project> {
+
+  override fun apply(project: Project) {
+    project.apply(ImmutableMap.of("plugin", "com.android.library"))
+    project.apply(ImmutableMap.of("plugin", LicenseResolverPlugin::class.java))
+    project.apply(ImmutableMap.of("plugin", "com.github.sherter.google-java-format"))
+    project.extensions.getByType(GoogleJavaFormatExtension::class.java).toolVersion = "1.10.0"
+    val firebaseLibrary =
+      project.extensions.create(
+        "firebaseLibrary",
+        FirebaseLibraryExtension::class.java,
+        project,
+        LibraryType.ANDROID
+      )
+    val android = project.extensions.getByType<LibraryExtension>()
```

Comment:
There's a lot going on with `android` here. tbh, there's a lot going on in `apply` in general. Thoughts on moving android configuration to its own method?

Reply:
Moved it out in both methods for consistency